sdk/storage/Azure.Storage.Files.Shares/src/ShareDirectoryClient.cs

```diff
@@ -1837,6 +1841,71 @@ await GetSubdirectoryClient(subdirectoryName)
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+        [ForwardsClientCalls]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual Response<ShareFileClient> CreateFile(
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            string fileName,
+            long maxSize,
+            ShareFileHttpHeaders httpHeaders,
+            Metadata metadata,
+            FileSmbProperties smbProperties,
+            string filePermission,
+            CancellationToken cancellationToken)
+        {
+            ShareFileClient file = GetFileClient(fileName);
```

Comment:
It would be nice to forward this on to the other `CreateFile` with the extra parameter instead of implementing it twice.

Reply:
(Can defer)